### ISSUE TITLE
"Reads are improperly paired" Error

### ISSUE THREAD

**Issue by imk1** [ENCODE-DCC/atac-seq-pipeline#207]
**Describe the bug**
When I run the pipeline, I sometimes get the following error:
atac.align RetryableFailure. SHARD_IDX=0, RC=1, JOB_ID=12002786, RUN_START=2019-12-17T17:47:05.891Z, RUN_END=2019-12-17T19:38:11.452Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-
a06b-132cf4b650a4/call-align/shard-0/execution/stderr
STDERR_CONTENTS=
Traceback (most recent call last):
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_task_trim_adapter.py", line 232, in <module>
    main()
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_task_trim_adapter.py", line 200, in main
    args.out_dir)
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_task_trim_adapter.py", line 128, in trim_adapter_pe
    run_shell_cmd(cmd)
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_lib_common.py", line 319, in run_shell_cmd
    raise Exception(err_str)
Exception: PID=117321, PGID=117321, RC=1
STDERR=cutadapt: error: Error in sequence file at unknown line: Reads are improperly paired. There are more reads in file 1 than in file 2.
I have verified that the number of reads in the fastq files from the 2 ends are exactly the same.  In addition, when I run cutadapt on the exact same files using the exact same settings that are used in the pipeline, cutadapt runs to completion without any problems.  Do you have any idea what might be causing this error and how I might be able to fix it?  Thanks!

**OS/Platform**
- OS/Platform: Linux cluster with Centos 6
- Conda version: 4.7.12
- Pipeline version: v1.5.4
- Caper version: 0.5.1

**Caper configuration file**
Paste contents of `~/.caper/default.conf`.
[defaults]
backend=slurm
slurm-partition=pfen1
slurm-account=ikaplow
java-heap-run=4G
tmp-dir=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/caper_tmp
no-file-db=True
ip=localhost
port=8000
show-completed-task=True

**Input JSON file**
Paste contents of your input JSON file.
{
 "atac.genome_tsv" : "/home/ikaplow/RegulatoryElementEvolutionProject/GenomeDataPipelines/mm10/mm10.tsv",
 "atac.fastqs_rep1_R1" : [ "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP462Liv/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L2_1.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP462Liv/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_1.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP462Liv/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L4_1.fq.gz" ],
 "atac.fastqs_rep1_R2" : [ "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP462Liv/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L2_2.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP462Liv/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_2.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP462Liv/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L4_2.fq.gz" ],
 "atac.fastqs_rep2_R1" : [ "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP465Liv/AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L2_1.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP465Liv/AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L3_1.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP465Liv/AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L4_1.fq.gz" ],
 "atac.fastqs_rep2_R2" : [ "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP465Liv/AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L2_2.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP465Liv/AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L3_2.fq.gz", "/projects/MPRA/bats/Rousettus_aegyptiacus/fastqs/128.120.88.251/H202SC19103078/Rawdata/AP465Liv/AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L4_2.fq.gz" ],
 "atac.pipeline_type" : "atac",
 "atac.paired_end" : true,
 "atac.auto_detect_adapter" : true,
 "atac.multimapping" : 0,
 "atac.enable_xcor" : false,
 "atac.cap_num_peak" : 300000,
 "atac.filter_chrs" : ["chrM", "MT"],
 "atac.smooth_win" : 150,
 "atac.enable_idr" : true,
 "atac.idr_thresh" : 0.1,
 "atac.enable_compare_to_roadmap" : true,
 "atac.enable_preseq" : true,
 "atac.align_mem_mb" : 40000,
 "atac.title" : "LiverPfenning",
 "atac.description" : "ATAC-seq data from mouse liver from Pfenning Lab"
}

**Error log**
Caper automatically runs a troubleshooter for failed workflows. If it doesn't then get a `WORKFLOW_ID` of your failed workflow with `caper list` or directly use a `metadata.json` file on Caper's output directory.
```
$ caper debug [WORKFLOW_ID_OR_METADATA_JSON_FILE]
```
Found failures:
[
    {
        "message": "Workflow failed",
        "causedBy": [
            {
                "message": "The job was aborted from outside Cromwell",
                "causedBy": []
            }
        ]
    }
]

atac.read_genome_tsv Done. SHARD_IDX=-1, RC=0, JOB_ID=12002785, RUN_START=2019-12-17T17:46:22.163Z, RUN_END=2019-12-17T17:46:59.992Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-read_genome_tsv/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-read_genome_tsv/execution/stderr
STDERR_CONTENTS=

atac.filter Done. SHARD_IDX=1, RC=0, JOB_ID=12006910, RUN_START=2019-12-18T17:03:07.831Z, RUN_END=2019-12-19T01:18:06.578Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-filter/shard-1/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-filter/shard-1/execution/stderr
STDERR_CONTENTS=

atac.align RetryableFailure. SHARD_IDX=0, RC=1, JOB_ID=12002786, RUN_START=2019-12-17T17:47:05.891Z, RUN_END=2019-12-17T19:38:11.452Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/stderr
STDERR_CONTENTS=
Traceback (most recent call last):
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_task_trim_adapter.py", line 232, in <module>
    main()
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_task_trim_adapter.py", line 200, in main
    args.out_dir)
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_task_trim_adapter.py", line 128, in trim_adapter_pe
    run_shell_cmd(cmd)
  File "/home/ikaplow/anaconda2/envs/encode-atac-seq-pipeline/bin/encode_lib_common.py", line 319, in run_shell_cmd
    raise Exception(err_str)
Exception: PID=117321, PGID=117321, RC=1
STDERR=cutadapt: error: Error in sequence file at unknown line: Reads are improperly paired. There are more reads in file 1 than in file 2.
STDOUT=This is cutadapt 2.5 with Python 3.7.0
Command line parameters: -e 0.1 -m 5 -a CTGTCTCTTATA -A CTGTCTCTTATA /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/inputs/1076795402/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_1.fq.gz /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/inputs/1076795402/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_2.fq.gz -o AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_1.trim.fastq.gz -p AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_2.trim.fastq.gz
Processing reads on 1 core in paired-end mode ...
ln: accessing `/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/*.bam': No such file or directory
ln: accessing `/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/*.bai': No such file or directory
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/script: fork: retry: Resource temporarily unavailable
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/script: fork: retry: Resource temporarily unavailable
ln: accessing `/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/*.samstats.qc': No such file or directory
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/script: fork: retry: Resource temporarily unavailable
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/script: fork: retry: Resource temporarily unavailable
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/script: fork: retry: Resource temporarily unavailable
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/script: fork: retry: Resource temporarily unavailable
ln: accessing `/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/non_mito/*.samstats.qc': No such file or directory
ln: accessing `/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/execution/*.read_length.txt': No such file or directory

atac.align Done. SHARD_IDX=0, RC=0, JOB_ID=12002828, RUN_START=2019-12-17T19:38:11.833Z, RUN_END=2019-12-19T03:43:40.495Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/attempt-2/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/attempt-2/execution/stderr
STDERR_CONTENTS=

atac.align Done. SHARD_IDX=1, RC=0, JOB_ID=12002787, RUN_START=2019-12-17T17:47:07.862Z, RUN_END=2019-12-18T17:02:58.935Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-1/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-1/execution/stderr
STDERR_CONTENTS=

atac.preseq Failed. SHARD_IDX=1, RC=None, JOB_ID=12006911, RUN_START=2019-12-18T17:03:09.837Z, RUN_END=2019-12-18T21:07:33.386Z, STDOUT=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/execution/stdout, STDERR=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/execution/stderr
STDERR_CONTENTS=
Picked up _JAVA_OPTIONS: -Djava.io.tmpdir=/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/tmp.8147f059
INFO    2019-12-18 15:45:23     EstimateLibraryComplexity

********** NOTE: Picard's command line syntax is changing.
**********
********** For more information, please see:
********** https://github.com/broadinstitute/picard/wiki/Command-Line-Syntax-Transition-For-Users-(Pre-Transition)
**********
********** The command line looks like this in the new syntax:
**********
**********    EstimateLibraryComplexity -INPUT ./AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L2_1.trim.merged.no_rg.bam -OUTPUT AP465Liv_CKDL190142778-1a-N703-N508_HMHTYDSXX_L2_1.trim.merged.picardcomplexity.qc -USE_JDK_DEFLATER TRUE -USE_JDK_INFLATER TRUE -VERBOSITY ERROR -QUIET TRUE
**********

slurmstepd: *** JOB 12006911 CANCELLED AT 2019-12-18T15:47:00 DUE TO TIME LIMIT on compute-1-38 ***
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/execution/script: line 1: 344293 Terminated              tee '/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/execution/stdout' < "$out60d6e960"
/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/execution/script: line 1: 344294 Terminated              tee '/projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-preseq/shard-1/execution/stderr' < "$err60d6e960" 1>&2

**Comment by leepc12** [comment]
```
-e 0.1 -m 5 -a CTGTCTCTTATA -A CTGTCTCTTATA /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/inputs/1076795402/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_1.fq.gz /projects/pfenninggroup/machineLearningForComputationalBiology/regElEvoGrant/MouseDNase/LiverPfenning_ATAC_out/atac/60d6e960-c825-46df-a06b-132cf4b650a4/call-align/shard-0/inputs/1076795402/AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_2.fq.gz -o AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_1.trim.fastq.gz -p AP462Liv_CKDL190142778-1a-N702-N508_HMHTYDSXX_L3_2.trim.fastq.gz

**Comment by imk1** [comment]
That ran to completion successfully.

**Comment by leepc12** [comment]
Can you append ` 1> o 2> e` to that command line and re-run? 
Also, run the following right after that command line ran to completion. This is to check the return code.
```
echo $?
```
Then upload `o` and `e` files here.

Is `CTGTCTCTTATA` a correct adapter for these FASTQs?

This can be helpful. https://github.com/marcelm/cutadapt/issues/199

**Comment by imk1** [comment]
I actually looked at that issue you posted before posting this issue to see if it would help, and it did not help.

I ended up re-running the pipeline, and it has now successfully trimmed and aligned the reads.  I am not sure why it got an error before and does not get one now, as the only thing that I have changed is turning off preseq.  Do not worry about looking into this for me alone, as things are now working for me.

Thanks for helping!
